### PR TITLE
Log lightly handled errors

### PR DESCRIPTION
**Pull request by emlun**
I went through all `except` statements and added error logging where it seemed appropriate. My basic rules of thumb for when this was not needed were:

- If the `except` block (or the code following a `pass` statement) attempts to recover from the error
- In `except` block branches that raise new errors
- If the `except` block is on the CLI level and appropriate feedback is already given

### CODE REVIEW

File: ykman/cli/oath.py [hunk 1]
```diff
@@ -474,7 +478,8 @@ def ensure_validated(ctx, prompt='Enter your password', remember=False):
             try:
                 controller.validate(a2b_hex(keys[controller.id]))
                 return
-            except Exception:
+            except Exception as e:
+                logger.debug('Error', exc_info=e)
                 del keys[controller.id]
```
Comment (emlun) [inline ykman/cli/oath.py]: What does line 483 actually do in practice?

File: ykman/cli/slot.py [hunk 2]
```diff
@@ -54,7 +58,8 @@ def inner(ctx, param, val):
                                      callback=lambda c, p, v: int(v))
 
 
-def _failed_to_write_msg(ctx):
+def _failed_to_write_msg(ctx, exc_info):
+    logger.error('Failed to write message to driver', exc_info=exc_info)
```
Comment (dagheyman) [inline ykman/cli/slot.py]: It didn't fail to write a message, the function name is referring to that it failed to write and should give the user a message. So I think the log error message should be something like "Failed to write to the device".

File: ykman/driver_ccid.py [hunk 3]
```diff
@@ -102,8 +102,9 @@ def __init__(self, connection, name):
         self._pid = _pid_from_name(name)
         try:
             self._read_serial()
-        except APDUError:
-            pass  # Can't read serial or version
+        except APDUError as e:
+            # Can't read serial or version
+            logger.error('Failed to read serial number', exc_info=e)
```
Comment (dagheyman) [inline ykman/driver_ccid.py]: Failed to read serial or version? Or is the comment outdated?

File: ykman/piv.py [hunk 4]
```diff
@@ -588,9 +592,8 @@ def set_mgm_key(self, new_key, touch=False, store_on_device=False):
                 self.put_data(
                     OBJ.PIVMAN_PROTECTED_DATA,
                     self._pivman_protected_data.get_bytes())
-            except APDUError:
-                # No pin provided, can't clear key..
-                pass
+            except APDUError as e:
+                logger.debug("No pin provided, can't clear key..", exc_info=e)
```
Comment (dagheyman) [inline ykman/piv.py]: pin -> PIN

**Comment by emlun** [comment]
I agree about that being a bit redundant. The reason I added those log statements was so there's some way to get the exception backtrace out, since the plain echo functions don't include that and the exception isn't re-thrown to be printed somewhere else.

File: ykman/cli/slot.py (hunk 2 above)
Comment (emlun) [reply ykman/cli/slot.py]: Ah, good catch!

File: ykman/driver_ccid.py (hunk 3 above)
Comment (emlun) [reply ykman/driver_ccid.py]: I wasn't sure, judging by the method name, so I left the comment in place. But it turns out the `_read_serial` method does indeed read both serial number and version, so I'll refactor that.

File: ykman/piv.py (hunk 4 above)
Comment (emlun) [reply ykman/piv.py]: 👍

**Comment by dagheyman** [comment]
Any feedback on this one @dainnilsson ?

**Pull request merged**